tracer/src/Datadog.Trace/Telemetry/TelemetryTransportManager.cs

```diff
@@ -46,6 +48,8 @@ private enum PushEvaluationResult
         FatalError
     }
 
+    public bool HasSentSuccessfully { get; private set; }
```

Comment:
Rather than make this public, and have the OneShotTelemetryReporter have to interpet this + the return value of `TryPushTelemetry`, I wonder if we should keep this private, and expose the `PushEvaluationResult` instead?

e.g. something like

```csharp
// New method, original signature
public async Task<bool> TryPushTelemetry(
        TelemetryData telemetryData,
        ICollection<TelemetryValue>? config,
        ICollection<DependencyTelemetryData>? dependencies,
        ICollection<IntegrationTelemetryData>? integrations)
    => PushTelemetry(telemetryData, config, dependencies, integrations) != PushEvaluationResult.FatalError;

// Original method, but exposes PushEvaluationResult
public async Task<PushEvaluationResult> PushTelemetry(
        TelemetryData telemetryData,
        ICollection<TelemetryValue>? config,
        ICollection<DependencyTelemetryData>? dependencies,
        ICollection<IntegrationTelemetryData>? integrations)
{
    if (_currentTransport >= _transports.Length)
    {
        return PushEvaluationResult.FatalError;
    }

    // ... etc, all the same, until
    return retryResult
}
```

Then we can avoid exposing this, and `OneShotTelemetryReporter.UploadTelemetryData` becomes something like:

```csharp
 private async Task<bool> UploadTelemetryData(TelemetryData telemetryData)
{
    while (true)
    {
        PushEvaluationResult result = await _telemetryTransportManager.PushTelemetry(telemetryData).ConfigureAwait(false);
        switch(result)
        {
            case PushEvaluationResult.Success: return true;
            case PushEvaluationResult.FatalError: return false;
            default;
              // transient error
              break;
        }
    }
}
```

Hmmm.... Now I'm not sure if that makes sense or not 😅

Reply:
D'oh! You're absolutely right, no reason for all that complexity. Reverted those changes and took your advice in https://github.com/DataDog/dd-trace-dotnet/commit/7ba4bb6bbbcb9992e7874a4625f5a783bbf05002.